Update data-retention-policy.rst (#4937)
mattermost/docs: source/comply/data-retention-policy.rst

By default, Mattermost stores all message history providing an unlimited search history to admins and end users.

In Mattermost Enterprise E20, you can set a global retention policy as well as custom retention policies to manage how long messages and file uploads are kept in Mattermost channels and Direct Messages

.. warning:: 
  Once a message or a file is deleted, the action is irreversible. Please use caution when setting up global or custom data retention policies.

Configuring a Global Data Retention Policy
-------------------------------------------

To set a global data retention policy in Mattermost v5.38 and later:

1. Go to **System Console > Compliance > Data Retention Policies**.
2. Select **Edit** from the menu located to the right of the **Global retention policy** table. 

> 3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specify how long to keep those messages. When a time is specified, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.

Reviewer: ```suggestion
3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then set how long to keep those messages. When a time is set, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```

> 4. Select a **File retention** option from the dropdown. Specify the number of days or ears to keep files. When a time is specified, uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**. The minimum retention period is one day. The global file policy deletes all files regardless of whether they're in a direct message, private, or public channel.

Reviewer: ```suggestion
4. Select a **File retention** option from the dropdown. Set the number of days or ears to keep files. When a time is set, uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**. The minimum retention period is one day. The global file policy deletes all files regardless of whether they're in a direct message, private, or public channel.
```

5. Under the **Policy log** section, select **Edit** to specify the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. 

Select **Save**. Messages and files older than the duration you set will be deleted at the specified server time, as applicable.

Configuring a Custom Data Retention Policy
-------------------------------------------

To set a custom data retention policy in Mattermost v5.38 and later:

1. Go to **System Console > Compliance > Data Retention Policies**.
2. Select **Add policy** to the right of the **Custom retention policies** table. 
3. Specify a name for your policy. 

> 4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specify how long to keep uploaded files. When a time is specified, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.

Reviewer: ```suggestion
4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then set how long to keep uploaded files. When a time is set, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```

.. warning:: 
  Once a message or a file is deleted, the action is irreversible. Please use caution when setting up a custom data retention policy.

Configuring a Global Data Retention Policy
-------------------------------------------

> To set a global data retention policy:

Reviewer: ```suggestion
To set a global data retention policy in Mattermost v5.38 and later:
```

> 1. Go to **System Console > Compliance > Data Retention Policy**.

Reviewer: ```suggestion
1. Go to **System Console > Compliance > Data Retention Policies**.
```

> 2. Select edit from the menu presented to the right of the **Global retention policy** table. 

Reviewer: ```suggestion
2. Select **Edit** from the menu located to the right of the **Global retention policy** table. 
```

> 3. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.

Reviewer: ```suggestion
3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specifying how long to keep those messages. When a time is specified, messages and file attachments older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```

> 3. Select a **File retention** option from the dropdown. Specify the number of days or ears to keep files. When a time is specified uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage** at the specified time. The minimum retention period is one day.

Reviewer: ```suggestion
3. Specify a global retention policy for files by selecting a **File retention** option from the dropdown, then specifying how long to keep uploaded files. When a time is specified, uploaded files older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**) at the specified time. The minimum retention period is one day.
```

Reviewer: We appear to be making a distinction between file attachments on a channel or DM message versus uploaded files. Is this intentional? If so, we may want to clearly emphasize the difference between the two types of "files" and under what conditions each global retention option will work.

Reviewer 2: the number 3 is used for two different bullet.

Reviewer 2: > We appear to be making a distinction between file attachments on a channel or DM message versus uploaded files. Is this intentional? If so, we may want to clearly emphasize the difference between the two types of "files" and under what conditions each global retention option will work.

Every file in the Channels product is attached to a post. The global file policy deletes all files (regardless of whether they're in a direct message, private, or public channel).

> 4. Set the start time of the daily scheduled data retention job under the **Policy log** section. Choose a time when fewer people are using your system. 

Reviewer: ```suggestion
4. Under the **Policy log** section, select **Edit** to specify the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. 
```

> Save the settings. Messages and files older than the duration you set will be deleted at the specified server time, if applicable.

Reviewer: ```suggestion
Select **Save**. Messages and files older than the duration you set will be deleted at the specified server time, as applicable.
```

Configuring a Custom Data Retention Policy
-------------------------------------------

> To set a custom data retention policy in Mattermost versions 5.38+:

Reviewer: ```suggestion
To set a custom data retention policy in Mattermost v5.38 and later:
```

> 1. Go to **System Console > Compliance > Data Retention Policy**.

Reviewer: ```suggestion
1. Go to **System Console > Compliance > Data Retention Policies**.
```

> 2. Select **Add policy** from the menu presented to the right of the **Custom retention policy** table. 

Reviewer: ```suggestion
2. Select **Add policy** to the right of the **Custom retention policies** table. 
```

> 3. Create a name for your policy. 

Reviewer: ```suggestion
3. Specify a name for your policy. 
```

> 4. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.

Reviewer: ```suggestion
4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specifying how long to keep uploaded files. When a time is specified, messages and file attachments older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```

> 5. Assign teams and channels to this policy by selecting **Add teams** and searching for a specific team or by selecting **Add channels** and searching for a specific channel. All channels will be included in the a policy assigned to a team only. 

Reviewer: ```suggestion
5. Assign teams and channels to this policy by selecting **Add teams** and searching for a specific team, or by selecting **Add channels** and searching for a specific channel. If only teams are specified, all channels for selected teams will be included in the a policy. 
```

> 6. Set the start time of the daily scheduled data retention job under the **Policy log** section. Choose a time when fewer people are using your system. If there is already a time set for global retention policies then this time will apply also to custom data retention policies. 

Reviewer: ```suggestion
6. Under the **Policy log** section, select **Edit** to specify the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. If a time is already set for a global retention policy, then the same time applies to custom data retention policies. 
```

> If no jobs are in progress and the job has stayed ``Pending`` for more than 2 minutes, then you may not have restarted your server after enabling the data retention policy. Restart your server and try again.

Reviewer: Is a server restart required when making global or custom data retention policies?

Author: @mkraft Can you confirm if a restart is still required for making changes to the data retention policies?

Reviewer 2: No it's not necessary anymore.

Author: Great Thanks!!
